force-app/main/adapter/in/sobjects/contact/ContactAdapter.cls

```diff
@@ -665,21 +665,36 @@ public inherited sharing class ContactAdapter extends fflib_SObjects2 {
             Address__c newAddressFromContact = contactAddressesByContact.get(contact);
             Address__c existingAddressFromContact = existingAddressesByAddress.get(newAddressFromContact);
 
-            // if found a match
-            if (contactAddressHasAddressMatch(existingAddressFromContact)) {
+            // Check if the contact has Address Override enabled
+            if (contact.is_Address_Override__c == true) {
```

Comment:
We typically capitalize the first character of a field name (not counting the namespace of course). 
```suggestion
            if (contact.Is_Address_Override__c == true) {
```